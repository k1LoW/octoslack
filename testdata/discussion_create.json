{
  "action": "created",
  "discussion": {
    "repository_url": "https://git.example.com/api/v3/repos/example/myrepo",
    "category": {
      "id": 1328,
      "node_id": "XXXXXXXXxxXXXXXxxxxxxXXXXXXX=",
      "repository_id": 239,
      "emoji": ":raised_hands:",
      "name": "Show and tell",
      "description": "Show off something you've made",
      "created_at": "2023-02-15T12:45:07.000+09:00",
      "updated_at": "2023-02-999T14:06:26.000+09:00",
      "slug": "show-and-tell",
      "is_answerable": false
    },
    "answer_html_url": null,
    "answer_chosen_at": null,
    "answer_chosen_by": null,
    "html_url": "https://git.example.com/example/myrepo/discussions/1042",
    "id": 13,
    "node_id": "XXXXXXxXXXxXXXXXX==",
    "number": 1042,
    "title": "octoslack has been released!",
    "user": {
      "login": "k1LoW",
      "id": 1234,
      "node_id": "XXXXxXXXXXXXXXXXxxx",
      "avatar_url": "https://avatars.git.example.com/u/1234?",
      "gravatar_id": "",
      "url": "https://git.example.com/api/v3/users/k1LoW",
      "html_url": "https://git.example.com/k1LoW",
      "followers_url": "https://git.example.com/api/v3/users/k1LoW/followers",
      "following_url": "https://git.example.com/api/v3/users/k1LoW/following{/other_user}",
      "gists_url": "https://git.example.com/api/v3/users/k1LoW/gists{/gist_id}",
      "starred_url": "https://git.example.com/api/v3/users/k1LoW/starred{/owner}{/repo}",
      "subscriptions_url": "https://git.example.com/api/v3/users/k1LoW/subscriptions",
      "organizations_url": "https://git.example.com/api/v3/users/k1LoW/orgs",
      "repos_url": "https://git.example.com/api/v3/users/k1LoW/repos",
      "events_url": "https://git.example.com/api/v3/users/k1LoW/events{/privacy}",
      "received_events_url": "https://git.example.com/api/v3/users/k1LoW/received_events",
      "type": "User",
      "site_admin": false
    },
    "state": "open",
    "locked": false,
    "comments": 0,
    "created_at": "2023-02-999T06:53:21Z",
    "updated_at": "2023-02-999T06:53:21Z",
    "author_association": "NONE",
    "active_lock_reason": null,
    "body": "`octoslack` is GitHub Webhooks to Slack Incoming Webhooks",
    "reactions": {
      "url": "https://git.example.com/api/v3/repos/example/myrepo/discussions/1042/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "timeline_url": "https://git.example.com/api/v3/repos/example/myrepo/discussions/1042/timeline"
  },
  "repository": {
    "id": 239,
    "node_id": "YYYYYYyyyyyyYYYYYYYY=",
    "name": "myrepo",
    "full_name": "example/myrepo",
    "private": true,
    "owner": {
      "login": "example",
      "id": 999,
      "node_id": "XXXXXXxxxXXXXXXXXXXXXX",
      "avatar_url": "https://avatars.git.example.com/u/999?",
      "gravatar_id": "",
      "url": "https://git.example.com/api/v3/users/example",
      "html_url": "https://git.example.com/example",
      "followers_url": "https://git.example.com/api/v3/users/example/followers",
      "following_url": "https://git.example.com/api/v3/users/example/following{/other_user}",
      "gists_url": "https://git.example.com/api/v3/users/example/gists{/gist_id}",
      "starred_url": "https://git.example.com/api/v3/users/example/starred{/owner}{/repo}",
      "subscriptions_url": "https://git.example.com/api/v3/users/example/subscriptions",
      "organizations_url": "https://git.example.com/api/v3/users/example/orgs",
      "repos_url": "https://git.example.com/api/v3/users/example/repos",
      "events_url": "https://git.example.com/api/v3/users/example/events{/privacy}",
      "received_events_url": "https://git.example.com/api/v3/users/example/received_events",
      "type": "Organization",
      "site_admin": false
    },
    "html_url": "https://git.example.com/example/myrepo",
    "description": "THIS IS MYREPO",
    "fork": false,
    "url": "https://git.example.com/api/v3/repos/example/myrepo",
    "forks_url": "https://git.example.com/api/v3/repos/example/myrepo/forks",
    "keys_url": "https://git.example.com/api/v3/repos/example/myrepo/keys{/key_id}",
    "collaborators_url": "https://git.example.com/api/v3/repos/example/myrepo/collaborators{/collaborator}",
    "teams_url": "https://git.example.com/api/v3/repos/example/myrepo/teams",
    "hooks_url": "https://git.example.com/api/v3/repos/example/myrepo/hooks",
    "issue_events_url": "https://git.example.com/api/v3/repos/example/myrepo/issues/events{/number}",
    "events_url": "https://git.example.com/api/v3/repos/example/myrepo/events",
    "assignees_url": "https://git.example.com/api/v3/repos/example/myrepo/assignees{/user}",
    "branches_url": "https://git.example.com/api/v3/repos/example/myrepo/branches{/branch}",
    "tags_url": "https://git.example.com/api/v3/repos/example/myrepo/tags",
    "blobs_url": "https://git.example.com/api/v3/repos/example/myrepo/git/blobs{/sha}",
    "git_tags_url": "https://git.example.com/api/v3/repos/example/myrepo/git/tags{/sha}",
    "git_refs_url": "https://git.example.com/api/v3/repos/example/myrepo/git/refs{/sha}",
    "trees_url": "https://git.example.com/api/v3/repos/example/myrepo/git/trees{/sha}",
    "statuses_url": "https://git.example.com/api/v3/repos/example/myrepo/statuses/{sha}",
    "languages_url": "https://git.example.com/api/v3/repos/example/myrepo/languages",
    "stargazers_url": "https://git.example.com/api/v3/repos/example/myrepo/stargazers",
    "contributors_url": "https://git.example.com/api/v3/repos/example/myrepo/contributors",
    "subscribers_url": "https://git.example.com/api/v3/repos/example/myrepo/subscribers",
    "subscription_url": "https://git.example.com/api/v3/repos/example/myrepo/subscription",
    "commits_url": "https://git.example.com/api/v3/repos/example/myrepo/commits{/sha}",
    "git_commits_url": "https://git.example.com/api/v3/repos/example/myrepo/git/commits{/sha}",
    "comments_url": "https://git.example.com/api/v3/repos/example/myrepo/comments{/number}",
    "issue_comment_url": "https://git.example.com/api/v3/repos/example/myrepo/issues/comments{/number}",
    "contents_url": "https://git.example.com/api/v3/repos/example/myrepo/contents/{+path}",
    "compare_url": "https://git.example.com/api/v3/repos/example/myrepo/compare/{base}...{head}",
    "merges_url": "https://git.example.com/api/v3/repos/example/myrepo/merges",
    "archive_url": "https://git.example.com/api/v3/repos/example/myrepo/{archive_format}{/ref}",
    "downloads_url": "https://git.example.com/api/v3/repos/example/myrepo/downloads",
    "issues_url": "https://git.example.com/api/v3/repos/example/myrepo/issues{/number}",
    "pulls_url": "https://git.example.com/api/v3/repos/example/myrepo/pulls{/number}",
    "milestones_url": "https://git.example.com/api/v3/repos/example/myrepo/milestones{/number}",
    "notifications_url": "https://git.example.com/api/v3/repos/example/myrepo/notifications{?since,all,participating}",
    "labels_url": "https://git.example.com/api/v3/repos/example/myrepo/labels{/name}",
    "releases_url": "https://git.example.com/api/v3/repos/example/myrepo/releases{/id}",
    "deployments_url": "https://git.example.com/api/v3/repos/example/myrepo/deployments",
    "created_at": "2022-08-06T05:07:49Z",
    "updated_at": "2022-11-22T03:20:06Z",
    "pushed_at": "2023-01-13T02:21:999Z",
    "git_url": "git://git.example.com/example/myrepo.git",
    "ssh_url": "git@git.example.com:example/myrepo.git",
    "clone_url": "https://git.example.com/example/myrepo.git",
    "svn_url": "https://git.example.com/example/myrepo",
    "homepage": "",
    "size": 8039,
    "stargazers_count": 33,
    "watchers_count": 33,
    "language": null,
    "has_issues": true,
    "has_projects": false,
    "has_downloads": true,
    "has_wiki": true,
    "has_pages": false,
    "forks_count": 0,
    "mirror_url": null,
    "archived": false,
    "disabled": false,
    "open_issues_count": 9991,
    "license": null,
    "allow_forking": true,
    "is_template": false,
    "web_commit_signoff_required": false,
    "topics": [

    ],
    "visibility": "internal",
    "forks": 0,
    "open_issues": 9991,
    "watchers": 33,
    "default_branch": "main"
  },
  "organization": {
    "login": "example",
    "id": 999,
    "node_id": "XXXXXXxxxXXXXXXXXXXXXX",
    "url": "https://git.example.com/api/v3/orgs/example",
    "repos_url": "https://git.example.com/api/v3/orgs/example/repos",
    "events_url": "https://git.example.com/api/v3/orgs/example/events",
    "hooks_url": "https://git.example.com/api/v3/orgs/example/hooks",
    "issues_url": "https://git.example.com/api/v3/orgs/example/issues",
    "members_url": "https://git.example.com/api/v3/orgs/example/members{/member}",
    "public_members_url": "https://git.example.com/api/v3/orgs/example/public_members{/member}",
    "avatar_url": "https://avatars.git.example.com/u/999?",
    "description": ""
  },
  "enterprise": {
    "id": 1,
    "slug": "example",
    "name": "Example, Inc.",
    "node_id": "XXXXxxXXXXXXXXXXXXXXXX",
    "avatar_url": "https://avatars.git.example.com/b/1?",
    "description": "",
    "website_url": "https://example.com",
    "html_url": "https://git.example.com/enterprises/example",
    "created_at": "2018-10-20T00:26:37Z",
    "updated_at": "2023-01-12T10:14:19Z"
  },
  "sender": {
    "login": "k1LoW",
    "id": 1234,
    "node_id": "XXXXxXXXXXXXXXXXxxx",
    "avatar_url": "https://avatars.git.example.com/u/1234?",
    "gravatar_id": "",
    "url": "https://git.example.com/api/v3/users/k1LoW",
    "html_url": "https://git.example.com/k1LoW",
    "followers_url": "https://git.example.com/api/v3/users/k1LoW/followers",
    "following_url": "https://git.example.com/api/v3/users/k1LoW/following{/other_user}",
    "gists_url": "https://git.example.com/api/v3/users/k1LoW/gists{/gist_id}",
    "starred_url": "https://git.example.com/api/v3/users/k1LoW/starred{/owner}{/repo}",
    "subscriptions_url": "https://git.example.com/api/v3/users/k1LoW/subscriptions",
    "organizations_url": "https://git.example.com/api/v3/users/k1LoW/orgs",
    "repos_url": "https://git.example.com/api/v3/users/k1LoW/repos",
    "events_url": "https://git.example.com/api/v3/users/k1LoW/events{/privacy}",
    "received_events_url": "https://git.example.com/api/v3/users/k1LoW/received_events",
    "type": "User",
    "site_admin": false
  }
}
