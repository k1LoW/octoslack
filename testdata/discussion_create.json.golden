POST /services/XXXXXxxxxxXXXXXX/XXXxxxxXXXXXXxxxxXXXXXX HTTP/1.1
Host: hooks.slack.com
Content-Type: application/json

{"blocks":[{"text":{"text":"Discussion created by k1LoW","type":"mrkdwn"},"type":"section"},{"text":{"text":"octoslack has been released!","type":"mrkdwn"},"type":"section"},{"text":{"text":"テストタイトル\\r\nテストタイトル","type":"mrkdwn"},"type":"section"},{"text":{"text":"\u003e テストタイトル\\r\n\u003e テストタイトル","type":"mrkdwn"},"type":"section"},{"text":{"text":"\u003e テスト固定\n","type":"mrkdwn"},"type":"section"}]}